R/simplify_network.R

```diff
@@ -78,10 +78,10 @@ simplify_network = function(rnet_y, parameters, region_boundary) {
   # Subsetting another dataset 'rnet_y' based on the spatial relation with 'rnet_merged_all_buffer'.
   # It selects features from 'rnet_y' that are within the boundaries of 'rnet_merged_all_buffer'.
   # rnet_y_subset = sf::st_intersection(rnet_yp, rnet_merged_all_projected_buffer)
-  browser()
+  # browser()
   rnet_yp_points = sf::st_point_on_surface(rnet_yp)
-  rnet_y_points_subset = rnet_y_points[rnet_merged_all_projected_buffer, ]
-  rnet_y_subset = rnet_yp[rnet_y_points_subset, ]
+  rnet_yp_points_subset = rnet_yp_points[rnet_merged_all_projected_buffer, ]
```

Comment:
need to check as lead to disconnected routes:
![image](https://github.com/user-attachments/assets/4a8f82d4-cd41-4b25-bd8d-24d67383df8a)

Reply:
Correct, thanks Zhao.